### PR TITLE
Using ament_target_dependencies where possible

### PR DESCRIPTION
**Pull request by ivanpauno**
Partially targets ros2/ros2#658

### CODE REVIEW

File: rcl/test/CMakeLists.txt [hunk 1]
```diff
@@ -107,38 +107,35 @@ function(test_target_function)
...
+    AMENT_DEPENDENCIES ${rmw_implementation} "osrf_testing_tools_cpp" "test_msgs"
   )
 
   rcl_add_custom_gtest(test_init${target_suffix}
     SRCS rcl/test_init.cpp
     ENV ${rmw_implementation_env_var} ${memory_tools_ld_preload_env_var}
     APPEND_LIBRARY_DIRS ${extra_lib_dirs}
     LIBRARIES ${PROJECT_NAME} osrf_testing_tools_cpp::memory_tools
-    AMENT_DEPENDENCIES ${rmw_implementation}
+    AMENT_DEPENDENCIES ${rmw_implementation} "osrf_testing_tools_cpp"
```
Comment (dirk-thomas) [inline rcl/test/CMakeLists.txt]: Why is this necessary where it wasn't used before?

Comment (ivanpauno) [reply rcl/test/CMakeLists.txt]: Corrected

**Comment by ivanpauno** [comment]
CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/6450/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/2821/)
* macOS [Build Status](https://ci.ros2.org/job/ci_osx/5296/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/6270/)

All failures are unrelated, so I'm merging
